repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala

```diff
@@ -152,7 +154,9 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
       "clusterId" -> clusterId,
       "nodeId" -> nodeId,
       "requestorIp" -> Option(stripHeaderParams(httpServletRequest.getHeader(CommonHttpHeader.X_FORWARDED_FOR.toString)))
-        .getOrElse(httpServletRequest.getRemoteAddr)
+        .getOrElse(httpServletRequest.getRemoteAddr),
+      "userResp" -> Option(stripHeaderParams(httpServletResponse.getHeader(X_PP_USER))).orNull,
+      "groupsResp" -> Option(stripHeaderParams(httpServletResponse.getHeader(X_PP_GROUPS))).orNull
```

Comment:
Kind of odd. You shouldn't need to wrap anything in an `Option` here, if null is an acceptable value.

That is, if `stripHeaderParams(...)` returns null, well, that's what you want.